pkg/cloud/services/ec2/natgateways.go

```diff
@@ -41,9 +43,21 @@ func (s *Service) reconcileNatGateways() error {
 
 	if len(s.scope.Subnets().FilterPrivate()) == 0 {
 		s.scope.V(2).Info("No private subnets available, skipping NAT gateways")
+		conditions.MarkFalse(
+			s.scope.AWSCluster,
+			infrav1.NatGatewaysReadyCondition,
+			infrav1.NatGatewaysReconciliationFailedReason,
+			clusterv1.ConditionSeverityWarning,
+			"No private subnets available, skipping NAT gateways")
```

Comment:
I'm wondering if marking the condition false here will lead to us never being able to consider the AWSCluster as "Ready" by conditions. I'm also not quite sure if we support deploying with "only" public subnets, @randomvariable what are your thoughts?

Reply:
Not sure we ever allow all-public subnets in managed mode, since we have this in subnets.go

```go
	// Check that we need at least 1 private and 1 public subnet after we have updated the metadata
	if len(subnets.FilterPrivate()) < 1 {
		record.Warnf(s.scope.AWSCluster, "FailedNoPrivateSubnet", "Expected at least 1 private subnet but got 0")
		return errors.New("expected at least 1 private subnet but got 0")
	}
	if len(subnets.FilterPublic()) < 1 {
		record.Warnf(s.scope.AWSCluster, "FailedNoPublicSubnet", "Expected at least 1 public subnet but got 0")
		return errors.New("expected at least 1 public subnet but got 0")
	}
```

Reply:
Gotcha, sounds like this shouldn't be an issue (at least for now) then :)